### DISCUSSION TITLE
command to get focus topmost parent (container)

### DISCUSSION THREAD

**Discussion by RayZ0rr** [i3/i3#4818] [Q&A] [answered]
What I want to do : 
```
bindsym $mod+o <command to topmost>, focus child, split v, spawn alacritty
bindsym $mod+u <command to topmost>, focus child, split h, spawn alacritty
```
Here alacritty serves as a placeholder. What I really want is: 
```
bindsym $mod+o <command to topmost>, focus child, split v, open
bindsym $mod+u <command to topmost>, focus child, split h, open
```
where `open` is `i3-msg open` with which I can spawn correct empty placeholders. I know its experimental, that's why I'll have to do more testing with it.

The layouts I want are shown below. But as of now they require multiple key presses that needs some care (more mental overhead): 
![empty_container1](https://user-images.githubusercontent.com/96843787/151310461-33276a00-db50-492c-93f5-454453284142.png)

![empty_container2](https://user-images.githubusercontent.com/96843787/151310484-652b2a2c-507e-4e2d-b441-e544b43a92cd.png)

![empty_container3](https://user-images.githubusercontent.com/96843787/151310492-a6993208-4e26-4445-8dbc-0d14e9a38fff.png)

**Comment by Airblader** [comment, marked as answer]
You haven't asked any question. What exactly are you looking for? What is "the topmost parent" to you?

**Reply by RayZ0rr** [reply]
Oh, I'm talking about the container that is focused when entering a new workspace.

**Reply by Airblader** [reply]
There is no command to focus the container that is focused when entering a new workspace because a workspace that is new doesn't have any containers in it that can be focused. :man_shrugging:

**Reply by RayZ0rr** [reply]
Okay, I'm having trouble saying this. In the third(last) image i posted, I'm looking at the command to focus on the parent container after which when I open a new terminal, the workspace will get 3 equally sized vertical columns. Terminal in 1st and 3rd column. Half terminal in 2nd (middle) column.

This is easy for the image i posted. But difficult if I was in a nested container terminal inside, say first terminal(left most).

**Reply by Airblader** [reply]
Such a command doesn't exist built-in, but can be scripted relatively easily using the IPC by traversing the tree upwards to the correct spot and returning that container's ID.

**Comment by RayZ0rr** [comment]
@Airblader I got a better description.
So, basically I want some thing like:
`bindsym <key> focus parent, focus parent, focus parent, focus parent...`
But I think it's better to just use `focus parent` till I get there.
